core/src/main/java/edu/wpi/grip/core/ProjectSettings.java

```diff
@@ -0,0 +1,85 @@
+package edu.wpi.grip.core;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Throwables;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * This object holds settings that are saved in project files.  This includes things like team numbers, which need to
+ * be preserved when deploying the project.
+ */
+@SuppressWarnings("PMD")
```

Comment:
# No, do not do this. Do it on the methods with the specific suppression. 

I won't merge this like this. Sorry.